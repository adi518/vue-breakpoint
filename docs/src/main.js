import 'babel-polyfill'
import Vue from 'vue'
import App from './App.vue'
import router from './router'
import VueAnalytics from 'vue-analytics'
import { VBreakpoint } from '@/components/Breakpoint'
import { Mixin as VBreakpointMixin } from 'vue-breakpoint-component'

Vue.config.productionTip = false

Vue.use(VueAnalytics, {
  id: 'UA-48463434-5',
  router
})

Vue.mixin(VBreakpointMixin)

new Vue({
  router,
  render: h => h(VBreakpoint, [App])
}).$mount('#app')
