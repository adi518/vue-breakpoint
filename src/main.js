// https://github.com/Akryum/v-tooltip/blob/master/src/index.js
import 'core-js/stable'
import 'regenerator-runtime/runtime'

import { extend } from './extend'
import Breakpoint from './breakpoint'

export default Breakpoint

export { extend, extend as Ctor } from './extend' // `Ctor` will be deprecated
export { default as VShowAt } from './show-at'
export { default as VHideAt } from './hide-at'
export { default as VBreakpoint } from './breakpoint'
export { default as VWithBreakpoint } from './with-breakpoint'

export const Plugin = {
  install(Vue, config) {
    const components = extend(config)
    Object.keys(components).forEach((name, Component) =>
      Vue.component(name, Component)
    )
  }
}

export const Install = Plugin // Will be deprecated

export const Mixin = {
  install(Vue) {
    Vue.mixin({
      inject: {
        $vBreakpoint: {
          from: 'breakpoint',
          default: undefined
        }
      }
    })
  }
}

export class Model {}

// Auto-install (Browser/Node)
let GlobalVue
if (typeof window !== 'undefined') {
  GlobalVue = window.Vue
} else if (typeof global !== 'undefined') {
  GlobalVue = global.Vue
}
if (GlobalVue) {
  GlobalVue.use(Plugin)
}
