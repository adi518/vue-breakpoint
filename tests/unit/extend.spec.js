import { extend } from '@/main'

const config = {
  breakpoints: {
    small: '(min-width: 476px)',
    medium: '(min-width: 668px)',
    large: '(min-width: 892px)',
    xlarge: '(min-width: 1100px)'
  }
}

describe('extend', () => {
  it('returns extended components', () => {
    expect(extend(config)).toMatchSnapshot()
  })
})
