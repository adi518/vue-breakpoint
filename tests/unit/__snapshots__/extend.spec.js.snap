// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extend returns extended components 1`] = `
Object {
  "VBreakpoint": Object {
    "beforeDestroy": [Function],
    "computed": Object {
      "flags": [Function],
      "noMatch": [Function],
    },
    "config": Object {
      "breakpoints": Object {
        "large": "(min-width: 892px)",
        "medium": "(min-width: 668px)",
        "small": "(min-width: 476px)",
        "xlarge": "(min-width: 1100px)",
      },
      "debounceTime": 100,
      "experimental": false,
    },
    "created": [Function],
    "data": [Function],
    "methods": Object {
      "getBreakpoint": [Function],
      "getScope": [Function],
      "getWindowAttrs": [Function],
      "log": [Function],
      "match": [Function],
    },
    "mounted": [Function],
    "name": "v-breakpoint",
    "props": Object {
      "debounceTime": Object {
        "type": [Function],
      },
      "value": Object {
        "type": [Function],
      },
    },
    "provide": [Function],
    "render": [Function],
  },
  "VHideAt": Object {
    "components": Object {
      "VShowAt": Object {
        "components": Object {
          "Fragment": Object {
            "abstract": true,
            "beforeDestroy": [Function],
            "data": [Function],
            "mounted": [Function],
            "name": "fragment",
            "props": Object {
              "show": Object {
                "default": true,
                "type": [Function],
              },
            },
            "template": <div :id="_uid">
  <slot></slot>
</div>,
            "watch": Object {
              "show": Object {
                "handler": [Function],
                "immediate": true,
              },
            },
          },
          "VBreakpoint": Object {
            "beforeDestroy": [Function],
            "computed": Object {
              "flags": [Function],
              "noMatch": [Function],
            },
            "config": Object {
              "breakpoints": Object {
                "large": "(min-width: 892px)",
                "medium": "(min-width: 668px)",
                "small": "(min-width: 476px)",
                "xlarge": "(min-width: 1100px)",
              },
              "debounceTime": 100,
              "experimental": false,
            },
            "created": [Function],
            "data": [Function],
            "methods": Object {
              "getBreakpoint": [Function],
              "getScope": [Function],
              "getWindowAttrs": [Function],
              "log": [Function],
              "match": [Function],
            },
            "mounted": [Function],
            "name": "v-breakpoint",
            "props": Object {
              "debounceTime": Object {
                "type": [Function],
              },
              "value": Object {
                "type": [Function],
              },
            },
            "provide": [Function],
            "render": [Function],
          },
        },
        "computed": Object {
          "computedShow": [Function],
        },
        "config": Object {
          "breakpoints": Object {
            "large": "(min-width: 892px)",
            "medium": "(min-width: 668px)",
            "small": "(min-width: 476px)",
            "xlarge": "(min-width: 1100px)",
          },
          "debounceTime": 100,
          "experimental": false,
        },
        "created": [Function],
        "data": [Function],
        "methods": Object {
          "onChange": [Function],
        },
        "mixins": Array [
          Object {
            "props": Object {
              "breakpoint": Object {
                "default": "",
                "type": [Function],
              },
            },
          },
        ],
        "name": "v-show-at",
        "props": Object {
          "forceShow": Object {
            "default": false,
            "type": [Function],
          },
        },
        "render": [Function],
      },
    },
    "computed": Object {
      "forceShow": [Function],
    },
    "config": Object {
      "breakpoints": Object {
        "large": "(min-width: 892px)",
        "medium": "(min-width: 668px)",
        "small": "(min-width: 476px)",
        "xlarge": "(min-width: 1100px)",
      },
      "debounceTime": 100,
      "experimental": false,
    },
    "data": [Function],
    "methods": Object {
      "onChange": [Function],
    },
    "mixins": Array [
      Object {
        "props": Object {
          "breakpoint": Object {
            "default": "",
            "type": [Function],
          },
        },
      },
    ],
    "name": "v-hide-at",
    "render": [Function],
  },
  "VShowAt": Object {
    "components": Object {
      "Fragment": Object {
        "abstract": true,
        "beforeDestroy": [Function],
        "data": [Function],
        "mounted": [Function],
        "name": "fragment",
        "props": Object {
          "show": Object {
            "default": true,
            "type": [Function],
          },
        },
        "template": <div :id="_uid">
  <slot></slot>
</div>,
        "watch": Object {
          "show": Object {
            "handler": [Function],
            "immediate": true,
          },
        },
      },
      "VBreakpoint": Object {
        "beforeDestroy": [Function],
        "computed": Object {
          "flags": [Function],
          "noMatch": [Function],
        },
        "config": Object {
          "breakpoints": Object {
            "large": "(min-width: 892px)",
            "medium": "(min-width: 668px)",
            "small": "(min-width: 476px)",
            "xlarge": "(min-width: 1100px)",
          },
          "debounceTime": 100,
          "experimental": false,
        },
        "created": [Function],
        "data": [Function],
        "methods": Object {
          "getBreakpoint": [Function],
          "getScope": [Function],
          "getWindowAttrs": [Function],
          "log": [Function],
          "match": [Function],
        },
        "mounted": [Function],
        "name": "v-breakpoint",
        "props": Object {
          "debounceTime": Object {
            "type": [Function],
          },
          "value": Object {
            "type": [Function],
          },
        },
        "provide": [Function],
        "render": [Function],
      },
    },
    "computed": Object {
      "computedShow": [Function],
    },
    "config": Object {
      "breakpoints": Object {
        "large": "(min-width: 892px)",
        "medium": "(min-width: 668px)",
        "small": "(min-width: 476px)",
        "xlarge": "(min-width: 1100px)",
      },
      "debounceTime": 100,
      "experimental": false,
    },
    "created": [Function],
    "data": [Function],
    "methods": Object {
      "onChange": [Function],
    },
    "mixins": Array [
      Object {
        "props": Object {
          "breakpoint": Object {
            "default": "",
            "type": [Function],
          },
        },
      },
    ],
    "name": "v-show-at",
    "props": Object {
      "forceShow": Object {
        "default": false,
        "type": [Function],
      },
    },
    "render": [Function],
  },
}
`;
